src/dapp/README.md

```diff
@@ -1,98 +1,366 @@
 
-contract A {
-  function isEven(uint x) returns (bool) {
-    return x % 2 == 0;
-  }
-}
+## Commands
 
-contract C is DSTest {
-  A a;
-  function setUp() public {
-    a = new A();
-  }
-  function test_isEven() external {
-    assertTrue(a.isEven(4));
-  }
-}
+### `dapp init`
+
+    dapp-init -- bootstrap a new dapp
+    Usage: dapp init
+
+Initializes the current directory to the default `dapp` structure,
+installing `ds-test` and creating two boilerplate contracts in the `src` directory.
+
+### `dapp build`
+
+    dapp-build -- compile the source code
+    Usage: dapp build [--extract]
+
+    --extract:  After building, write the .abi, .bin and .bin-runtime. Implies `--legacy`
+        files from the solc json into $DAPP_OUT. Beware of contract
+        name collisions. This is provided for compatibility with older
+        workflows.
+    --optimize: activate the solidity optimizer.
+    --legacy:   Compile using the `--combined-json` flag. Some options are
+        missing from this format. This is provided for compatibility with older
+        workflows.
+
+Compiles the contracts in the `src` directory. 
+The compiler options of the build are generated by the [`dapp mk-standard-json`](#dapp-mk-standard-json) command,
+which infers most options from the project structure. For more customizability, you can define your own configuration json
+by setting the file to the environment variable `DAPP_STANDARD_JSON`.
+
+By default, `dapp build` uses [`dapp remappings`](#dapp-remappings) to resolve Solidity import paths.
+
+You can override this with the `DAPP_REMAPPINGS` environment variable.
+
+### `dapp test`
+
+    Usage: dapp test [<options>]
+  
+    Options:
+        -v, --verbose             trace ouput for failing tests
+        -vv                       trace output for all tests including passes
+        --verbosity <number>      sets the verbosity to <number>
+        --fuzz-runs <number>      number of times to run fuzzing tests
+        --replay <string>         rerun a particular test case
+        -m, --match <string>      only run test methods matching regex
+
+    RPC options:
+        --rpc                     fetch remote state via ETH_RPC_URL
+        --rpc-url <url>           fetch remote state via <url>
+        --rpc-block <number>      block number (latest if not specified)
+
+    SMT options:
+        --smttimeout <number>     timeout passed to the smt solver in ms (default 30000)
+        --solver <string>         name of the smt solver to use (either "z3" or "cvc4")
+        --max-iterations <number> number of times we may revisit a particular branching point during symbolic execution
+
+
+dapp tests are written in Solidity using the `ds-test` module. To install it, run
+```sh
+dapp install ds-test
 ```
 
-Run `dapp test` to run the test suite.
+Every contract which inherits from `DSTest` will be treated as a test contract, and will be assumed to have a public `setUp()` function, which will be run before every test.
```

Comment:
```suggestion
Every contract which inherits from `DSTest` will be treated as a test contract, if it has a `setUp()` function, it will be run before every test.
```